File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/EnumSituacaoNotasPendetesIntegracaoMercadoLivre.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{

    public enum EnumSituacaoNotasPendetesIntegracaoMercadoLivre
    {
        Pendente = 0,
        Concluido = 1,
        Falha = 2,
        Todas = 3
    }

    public static class EnumSituacaoNotasPendetesIntegracaoMercadoLivreHelper
    {
        public static string ObterDescricao(this EnumSituacaoNotasPendetesIntegracaoMercadoLivre situacao)
        {
            switch (situacao)
            {
                case EnumSituacaoNotasPendetesIntegracaoMercadoLivre.Pendente: return "Pendente Download";
                case EnumSituacaoNotasPendetesIntegracaoMercadoLivre.Concluido: return "Concluído";
                case EnumSituacaoNotasPendetesIntegracaoMercadoLivre.Falha: return "Falha";
                case EnumSituacaoNotasPendetesIntegracaoMercadoLivre.Todas: return "Todas";
                default: return string.Empty;
            }
        }
    }

}

File: SGT.WebAdmin/ViewsScripts/Enumeradores/EnumSituacaoNotasPendetesIntegracaoMercadoLivre.js
var EnumSituacaoNotasPendetesIntegracaoMercadoLivreHelper = function () {
    this.Pendente = 0,
    this.Concluido = 1,
    this.Falha = 2,
    this.Todas = 3
};

EnumSituacaoNotasPendetesIntegracaoMercadoLivreHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "Pendente Download", value: this.Pendente },
            { text: "Concluído", value: this.Concluido },
            { text: "Falha", value: this.Falha },
            { text: "Todas", value: this.Todas }
        ];
    }
}

var EnumSituacaoNotasPendetesIntegracaoMercadoLivre = Object.freeze(new EnumSituacaoNotasPendetesIntegracaoMercadoLivreHelper());